File: seongkyu-lim/Programmers/step1/이상한문자만들기.py
def solution(s):
    answer = ''
    cnt = 0
    new_word = []
    for i in range(0, len(s)):
        if s[i] != ' ':
            if cnt % 2 == 0:
                new_word.append(s[i].upper())
                cnt += 1
            else:
                new_word.append(s[i].lower())
                cnt += 1
        else:
            if cnt != 0:
                answer = answer + ''.join(new_word)
            answer += ' '
            cnt = 0
            new_word = []
            print(answer)

        if i == len(s)-1:
            answer = answer + ''.join(new_word)
    return answer


s = 'Hello  eVeryone hi'

print(solution(s))

File: mmkser10/javascript/programmers/이상한_문자_만들기.js
function solution(s) {
    let lower = s.toLowerCase().split(" ");
    let upper = s.toUpperCase().split(" ");
    var answer = "";
    for(let i = 0 ; i < lower.length ; i++) {
        for(let j = 0 ; j < lower[i].length ; j++){
            j%2 === 0 ? answer = answer + upper[i].charAt(j) : answer = answer + lower[i].charAt(j);
        }
        if(i !== lower.length-1)
            answer = answer + " ";
    }
    return answer;
}

// another sol
function anotherSolutionA(s) {
    return s.split(" ").map(i => i.split('').map((j, key) => key % 2 === 0 ? j.toUpperCase() : j).join(' ')).join(' ')
}